### DISCUSSION TITLE
Arborescence de fichiers

### DISCUSSION THREAD

**Discussion by Clm-Roig** [GrottoCenter/grottocenter-front#350] [General]
Actuellement, on a l'arborescence suivante : 
![image](https://user-images.githubusercontent.com/20704943/159918494-5ab8e994-6a63-46b6-8b7d-cdb7ca5a6f20.png)

Ainsi que la page wiki suivante : 
[https://github.com/GrottoCenter/grottocenter-front/wiki/Development-guide-(to-improve-&-complete!)](https://github.com/GrottoCenter/grottocenter-front/wiki/Development-guide-(to-improve-&-complete!))

Ce n'est pas trop homogène et plusieurs développeurs se sont succédés avec des visions de React différentes. Je pense notamment avec la notion de smart et dumb composants qui n'est plus vraiment d'actualité dans React désormais.

Voici mes suggestions pour nettoyer / repenser tout ça : 
- `components/admin` n'a pas lieu d'être, les composants étant des briques simples réutilisables : ce n'est pas le cas ici. Il s'agit d'un vieux bout de code qui n'a jamais été utilisé de toute façon : à supprimer. 
- `components/homepage` devrait être migrer dans `pages/homepage`. Le dossier `advancedSearch` irait dans `components/appli`.
- `features` est synonyme de `components/appli` selon moi. Ce sont des composants métiers complexes, propres à Grottocenter dans les deux cas. A déplacer.
- `containers` est un relicat de l'ancienne façon de coder du React. Un jour il faudra migrer le code vers ce qui se fait actuellement (composant fonctionnel pur). Pour le moment on garde mais on ne développe plus rien dedans.

Au final on se retrouverait avec : 
- components 
    - appli
    - common 
    (et rien d'autre)
- containers (déprécié, à migrer un jour)
- pages

J'attends vos avis :slightly_smiling_face:

**Comment by Clm-Roig** [comment]
@ClemRz, @maximenathan, @bsoufflet si vous passez par ici, votre avis m'est précieux !

**Reply by ClemRz** [reply]
Cela me parait bien. Ceci dit, ca fait longtemps que je ne fait plus de React ;)

**Comment by Clm-Roig** [comment]
Up, j'aimerais bien qu'on avance là-dessus pour ranger un peu le front :slightly_smiling_face: 
@maximenathan @bsoufflet qu'en pensez-vous ?

**Reply by bsoufflet** [reply]
Cela me parait bien également. Cependant mon expérience en React n'est pas suffisante pour que tu puisses tout baser sur mon avis ;)
Je serais d'avis de mettre un README dans containers juste pour expliquer que ce dossier est deprecated et que à la place il faut faire...
ET pourquoi pas 2 README minimalistes dans COmponents et pages pour expliqer rapidement leur organisation (ou sinon mettre cette explication dans le README general?).
Merci pour le refactor en tout cas !

**Reply by bsoufflet** [reply]
@Clm-Roig

**Reply by Clm-Roig** [reply]
ça marche, merci pour le retour. D'accord pour les README, c'est également mentionner dans le Wiki si besoin.
Je vais essayer de faire ça rapidement avant que les étudiants ne se mettent au travail.

**Comment by Clm-Roig** [comment]
Je ferme car fixé par #394, merci pour vos retours !
